pkg/config/constants/constants.go

```diff
@@ -199,6 +198,10 @@ const (
 
 	// AmbientUseWaypoint is the annotation used to specify which waypoint should be used for a given pod, service, etc...
 	AmbientUseWaypoint = "istio.io/use-waypoint"
+	// AmbientWaypointForTrafficType is the annotation used to specify which traffic should go through the intended waypoint.
+	// This annotation can be applied to both services and workloads. Valid traffic types are "service", "workload", "all",
```

Comment:
Isn't it applied to gateway?

Reply:
Yes, that's correct. The annotation is applied to the waypoint. The traffic type could be workload or service. Will update.